core/src/main/java/org/frankframework/parameters/NumberParameter.java

```diff
@@ -115,26 +115,24 @@ protected Number getValueAsType(Object request, boolean namespaceAware) throws P
 			if (request instanceof Number number) {
 				return number;
 			}
-			final Message requestMessage = Message.asMessage(request);
-			log.debug("Parameter [{}] converting result [{}] to number decimalSeparator [{}] groupingSeparator [{}]", this::getName, ()->requestMessage, decimalFormatSymbols::getDecimalSeparator, decimalFormatSymbols::getGroupingSeparator);
+			log.debug("Parameter [{}] converting result [{}] to number decimalSeparator [{}] groupingSeparator [{}]", this::getName, ()->request, decimalFormatSymbols::getDecimalSeparator, decimalFormatSymbols::getGroupingSeparator);
 			DecimalFormat decimalFormat = new DecimalFormat();
 			decimalFormat.setDecimalFormatSymbols(decimalFormatSymbols);
-			try {
+			try (Message requestMessage = Message.asMessage(request)) {
 				return decimalFormat.parse(requestMessage.asString());
 			} catch (ParseException e) {
-				throw new ParameterException(getName(), "Parameter [" + getName() + "] could not parse result [" + requestMessage + "] to number decimalSeparator [" + decimalFormatSymbols.getDecimalSeparator() + "] groupingSeparator [" + decimalFormatSymbols.getGroupingSeparator() + "]", e);
+				throw new ParameterException(getName(), "Parameter [" + getName() + "] could not parse result [" + request + "] to number decimalSeparator [" + decimalFormatSymbols.getDecimalSeparator() + "] groupingSeparator [" + decimalFormatSymbols.getGroupingSeparator() + "]", e);
```

Comment:
moet het niet requestMessage blijven? Hieronder ook?

Reply:
Die is nu niet meer in scope. Nu krijg je de `.toString()` op het oorspronkelijke object, anders op de Message. Ik denk niet dat je daarmee veel meer informatie krijgt, tenzij de Message het type van `request` had veranderd.